### PR TITLE
Add a build script to validate features

### PR DESCRIPTION
**Pull request by jessebraham**
Ensures that exactly 1 chip and exactly 1 communication method are enabled.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,17 +3,18 @@ name = "esp-println"
...
 
-uart = []
-jtag_serial = [] # C3 + S3
-rtt = []
-
-default = [ "uart" ]
+# You must enable exactly 1 of the below features to enable to intended
+# communication method (note that "uart" is enabled by default):
+uart        = []
+jtag_serial = [] # C3 + S3 only!
```
Comment (MabezDev) [inline Cargo.toml:19]: Maybe we should check this in the build script too?

Comment (jessebraham) [reply Cargo.toml:19]: Sorry what do you mean? It already checks that exactly one of `uart`, `jtag_serial`, or `rtt` are enabled.

Comment (MabezDev) [reply Cargo.toml:19]: Sorry! I meant that jtag_serial is only allowed on the C3 or S3.

Comment (jessebraham) [reply Cargo.toml:19]: All good, I figured it out eventually 😂 I've updated the script.

**Review by MabezDev — APPROVED** [review]
LGTM!

**Review by bjoernQ — APPROVED** [review]
LGTM

**Pull request merged**